include/deal.II/matrix_free/cuda_fe_evaluation.h

```diff
@@ -132,20 +132,36 @@ namespace CUDAWrappers
 
     /**
      * Return the value of a finite element function at quadrature point
-     * number @p q_point after a call to @p evalue(true,...).
+     * number @p q_point after a call to @p evaluate(true,...).
      */
     __device__ value_type
                get_value(const unsigned int q_point) const;
 
+    /**
+     * Return the value of a finite element function at degree of freedom
+     * @p dof after a call to integrate() or before a call to evaluate().
+     */
+    __device__ value_type
+               get_dof_value(const unsigned int dof) const;
```

Comment:
The idea would be to extract some value computed by a sequence of FEEvaluation::evaluate(), some quadrature point operation, and FEEvaluation::integrate(). The typical case is, as @masterleinad mentions, the computation of the diagonal by simply computing the columns of the cell matrix by repeatedly applying the operator to the unit vectors. See the function `local_compute_diagonal` of [step-37](https://www.dealii.org/developer/doxygen/deal.II/step_37.html#LocalevaluationofLaplaceoperator) for an example. You then only keep one value. The efficiency topic of using a single lane of a CUDA block aside, this is a viable thing to do. Or do you have more specific objections to the current layout?

Reply:
I am asking because the last time I was talking with @masterleinad  about this, we didn't need use `get_dof_value()`. If you look at Karl's implementation (see [here](https://github.com/kalj/dealii-cuda/blob/c8b0d3178001e2e694ce92b74b5fa8af6800d9c0/laplace_operator_gpu.h#L391)), he doesn't need to introduce `get_dof_value`. So I just want to know what's the difference with Karl's implementation.

Reply:
Since we are accessing the same field (both in `MatrixFree::FEEvaluation` and in `CUDAWrappers::FEEvaluation`), we could use `get_value` instead of `get_dof_value` (apart from the `Assert`s. On the other hand, I consider it useful to differentiate between access to values stored for quadrature points and degrees of freedom. Furthermore, I like both interfaces to be as similar as (easily) possible.